### ISSUE TITLE
Comparing Row equality in DatasetComparer

### ISSUE THREAD

**Issue by MrPowers** [MrPowers/spark-fast-tests#8]
Option 1: Use the built in [org.apache.spark.sql.Row#equals method](https://github.com/holdenk/spark-testing-base/blob/0a235cf9cd26b1152569618b110f0f73eb6d4181/src/main/1.6/scala/com/holdenkarau/spark/testing/DatasetSuiteBase.scala#L56)? 

Option 2: Use something like the [approxEquals method](https://github.com/holdenk/spark-testing-base/blob/7191b78ce8160f5dc9e44a3a5751f771bac66848/src/main/2.0/scala/com/holdenkarau/spark/testing/DataFrameSuiteBase.scala#L177).

@snithish - I think we should stick with Option 1 and delete the [RowComparer class](https://github.com/MrPowers/spark-fast-tests/blob/master/src/main/scala/com/github/mrpowers/spark/fast/tests/RowComparer.scala).  All the tests are passing with Option 1 and it's less complex.  Thoughts?

**Comment by snithish** [comment]
@MrPowers Option 3: Have a 
```scala
 def assertLargeDatasetEquality[T: ClassTag](actualDS: Dataset[T],
                                              expectedDS: Dataset[T],
                                              equals: (T, T) => Boolean)
```
So users could pass in their custom equality implementation and 
[equality check](https://github.com/MrPowers/spark-fast-tests/blob/master/src/main/scala/com/github/mrpowers/spark/fast/tests/DatasetComparer.scala#L95) would become `equals(o1, o2)`. Thoughts?

**Comment by MrPowers** [comment]
@snithish - Option 3 sounds really interesting!

Could we make `Row#equals` the default equality function and let users override it if they'd like to customize it?

Are you interested in starting a pull request for this one?

**Comment by snithish** [comment]
@MrPowers I have started working on it. Will raise a PR ASAP

**Comment by snithish** [comment]
@MrPowers #10 by default uses the equals method on the object. To make it easier for simple comparisons we would convert 
```scala
def areRowsEqual(r1: Row, r2: Row, tol: Double): Boolean
```
in RowComparer as 
```scala
def areRowsEqual(tol: Double)(r1: Row, r2: Row): Boolean
```
to be in sync with spark-testing-base, thereby giving users flexibility to use their own delta . Thoughts?

**Comment by MrPowers** [comment]
Merged in #10
